sd_tracking_utils.py

```diff
@@ -47,3 +53,225 @@ def map_tracker_ids(robot_mapping, id_list):
+
+        Returns:
+        --------
+            - torch.Tensor: A concatenated tensor containing all the input data with the missing ID.
+        """
+        return (
+            torch.cat((xyxy_tensor[0].reshape(1),
+                       xyxy_tensor[1].reshape(1),
+                       xyxy_tensor[2].reshape(1),
+                       xyxy_tensor[3].reshape(1),
+                       torch.tensor([missing_id]).reshape(1),
+                       conf_tensor.reshape(1),
+                       cls_tensor.reshape(1))))
+
+    @staticmethod
+    def process_result_list(result: Results, bad_id: int, missing_id: int) -> list:
+        """
+        ## Process a YOLO Results object, filtering out data with a specific bad ID and replacing it with a the data with the missing ID.
+        ----
+        Args:
+        ----
+            - result (Results): The Results object containing data to be processed.
+            - bad_id (int): The ID of the data to be filtered out.
+            - missing_id (int): The ID to replace the filtered data with.
+
+        Returns:
+        ----
+            - list: A list of processed data tensors with the missing ID.
+
+        """
+        data = []
+
+        for idx, tensor in enumerate(result.boxes.id):
+            try:
+                if SdUtils.is_bad_id(result.boxes.id[idx], bad_id):
+                    print("Found bad id: %s" % bad_id)
+
+                    data.append(SdUtils.process_data_with_bad_id(
+                        result.boxes.xyxy[idx], missing_id, result.boxes.conf[idx], result.boxes.cls[idx]))
+                    print(f'Appended Data Without Bad Id {data[-1]}')
+                else:
+                    data.append(SdUtils.process_data(
+                        result.boxes.xyxy[idx], result.boxes.id[idx], result.boxes.conf[idx], result.boxes.cls[idx]))
+                    print(f'Appended Data {data[-1]}')
+
+            except Exception as e:
+                print(f'Error {e}')
+
+        return data
+
+    @staticmethod
+    def update_yolo_result(data: list, result: Results) -> None:
+        """
+        ## Update a Results object with the processed data.
+        ----
+        Args:
+        ----
+            - data (list): A list of processed data tensors.
+            - result (Results): The YOLO Results object to be updated.
+        """
+        modified_data = torch.stack(data)
+
+        result.update(modified_data)
+
+    @staticmethod
+    def remove_false_detections_from_list(data: list, result: Results) -> list:
+        """
+        ## Remove false detections from a list of data.
+        ----
+        Args:
+        ----
+            - data (list): A list of data tensors to be updated.
+            - result (Results): The YOLO Results object containing detection data.
+
+        Returns:
+        ----
+            - list: The updated list of data tensors.
+
+        """
+        for idx in range(6):
```

Comment:
what is 6? write a Constant for the number of robots, what will happend if we will go back to the corn game?